src/edu/csus/ecs/pc2/exports/ccs/ResultsFile.java

```diff
@@ -292,30 +207,56 @@ public String[] createTSVFileLines(IInternalContest contest, Group group, String
+
+            if (record.getNumberSolved() > 0) {
+                if (finalizeData.isUseWFGroupRanking()) {
+                    if (record.getNumberSolved() >= highestHonorSolvedCount) {
+                        isHighestHonor = true;
+                    } else if (record.getNumberSolved() >= highHonorSolvedCount) {
+                        isHighHonor = true;
+                    } else if (record.getNumberSolved() >= median) {
+                        isHonor = true;
+                    }
+                }
+                else if (record.getNumberSolved() >= median) {
+                    isHonor = true;
+                }
             }
-            String award = getAwardMedal(record.getRankNumber(), finalizeData, ranked);
+
+            String award = getAwardMedal(record.getRankNumber(), finalizeData, isHighestHonor, isHighHonor, isHonor);
             String rank = "";
             if (!"honorable".equalsIgnoreCase(award)) {
-                if(finalizeData.isUseWFGroupRanking()) {
-                    if (realRank > lastMedalRank && (lastSolvedNum != record.getNumberSolved())) {
+                if (finalizeData.isUseWFGroupRanking() && realRank > lastMedalRank) {
+                    if (record.getNumberSolved() != lastSolvedNum) {
                         lastSolvedNum = record.getNumberSolved();
                         rankNumber = realRank;
-                        record.setRankNumber(realRank);
-                    } else if (realRank > lastMedalRank && lastSolvedNum == record.getNumberSolved() && lastSolvedNum > 0) {
-                        record.setRankNumber(rankNumber);
                     }
+                    record.setRankNumber(rankNumber);
                 }
                 rank = Integer.toString(record.getRankNumber());
             }
-            lines.addElement(reservationId + TAB //
-                    + rank + TAB //
-                    + award + TAB  //
-                    + record.getNumberSolved() //
-                    + TAB + record.getPenaltyPoints() + TAB //
-                    + record.getLastSolved());
+
+            if (isTSV) {
+                lines.addElement(reservationId + TAB //
+                        + rank + TAB //
+                        + award + TAB  //
+                        + record.getNumberSolved() + TAB //
+                        + record.getPenaltyPoints() + TAB //
+                        + record.getLastSolved());
+            } else {
+                // teamId,rank,medalCitation,problemsSolved,totalTime,lastProblemTime,siteCitation,citation
+                lines.addElement(reservationId + COMMA //
+                        + rank + COMMA //
+                        + award + COMMA //
+                        + record.getNumberSolved() + COMMA //
+                        + record.getPenaltyPoints() + COMMA //
+                        + record.getLastSolved() + COMMA // then siteCitation
+                        + COMMA);  // then citation
```

Comment:
I do not understand why the comment at the start of this `else` block indicates trailing values of `siteCitation` and `citation` for each record in a CSV file (as does the "header" line written above, around new line 151), but this block of code doesn't add those values to the file.

Reply:
I'm not sure what does `siteCitation` and `citation` mean here and what is the difference between the two. If it's something like 1st ranked teams from each region (like DOMjudge does) then I've plans on adding them as a separate a Feature request. I can remove these fields for the time being from here.

Reply:
The comment indicates what the CMS will actually accept.  In fact, the code DOES add those fields, but they are empty.  This is because they are not available.  At some point someone added code to produce the CSV file (which is what the CMS actually expects - remember, the CMS does NOT accept the TSV file!).  The generation of the CSV file was never done previously, as @clevengr points out, but, the code was there.  We now expose a report interface to generate the CSV file.  The **siteCitation**, as @SamanwaySadhu points out is for 1st ranked team from each region (group), and DOMjudge does include this in the TSV file (DOMjudge does not produce a CSV file).  The citation field is always blank - and we have no idea what is supposed to be there.  Perhaps this is a question for the CMS folks?  At some point, we should figure out how to get the "group" winner into BOTH the TSV and the CSV - we never did it for neither, and probably should.  This should be a separate Issue and addressed in a separate PR IMHO.  Does someone want to create an issue for this?  We should almost certainly address BEFORE PC2 becomes a primary.

Reply:
I agree: this should be a separate Issue and then have a separate PR to handle it.